icepyx/tests/test_Earthdata.py

```diff
@@ -0,0 +1,125 @@
+"""
+test_Earthdata.py (05/2021)
+"""
+import os
+import re
+import netrc
+import pytest
+import shutil
+import getpass
+import warnings
+import posixpath
+# from icepyx.core.Earthdata import Earthdata
+
+# PURPOSE: test different authentication methods
+@pytest.fixture(scope="module", autouse=True)
+def setup_earthdata():
+    # Before test - move .netrc file to a temporary place
+    netrc_file = os.path.join(os.path.expanduser('~'),'.netrc')
+    temprc_file = f"{netrc_file}.temp"
+    if os.access(netrc_file,os.F_OK):
+        shutil.move(netrc_file,temprc_file)
+    yield
+    # After test - move .netrc file back to original path
+    if os.access(temprc_file,os.F_OK):
+        shutil.move(temprc_file,netrc_file)
+    else:
+        os.remove(netrc_file)
+
+def capability_url(dataset,version):
+    return posixpath.join("https://n5eil02u.ecs.nsidc.org","egi",
+        "capabilities",f"{dataset}.{version}.xml")
+
+def test_environment(username,password,email):
+    url = capability_url("ATL06","003")
+    netrc_file = os.path.join(os.path.expanduser('~'),'.netrc')
+    assert not os.access(netrc_file,os.F_OK)
+    assert Earthdata(username,email,url).login(attempts=1)
+
+def test_netrc(username,password,email):
+    # append to netrc file and set permissions level
+    args = (username,'urs.earthdata.nasa.gov',password)
+    netrc_file = os.path.join(os.path.expanduser('~'),'.netrc')
+    with open(netrc_file,'a+') as f:
+        f.write('machine {1} login {0} password {2}\n'.format(*args))
+        os.chmod(netrc_file, 0o600)
+    url = capability_url("ATL06","003")
+    assert Earthdata(username,email,url).login(attempts=0)
+
+class Earthdata:
+    """
+    Mocks the icepyx Earthdata login
+
+    Parameters
+    ----------
+    uid : str
+        Earthdata Login user name (user ID).
+    email : str
+        Complete email address, provided as a string.
+    password : str
+        Password for Earthdata registration associated with the uid.
+    capability_url : str
+        URL required to access Earthdata
+
+    Returns
+    -------
+    success : bool
+        True if completed a successful "login", else False.
+    """
+
+    def __init__(
+        self,
+        uid,
+        email,
+        capability_url,
+        pswd=os.environ.get('EARTHDATA_PASSWORD'),
+    ):
+
+        assert isinstance(uid, str), "Enter your login user id as a string"
+        assert re.match(r"[^@]+@[^@]+\.[^@]+", email), \
+            "Enter a properly formatted email address"
+
+        self.netrc = None
+        self.uid = uid
+        self.email = email
+        self.capability_url = capability_url
+        self.pswd = pswd
+
+    def _start_session(self):
+        """
+        Checks supplied credentials versus environmental variables
+        """
+        EARTHDATA_USERNAME = os.environ.get('EARTHDATA_USERNAME')
+        EARTHDATA_PASSWORD = os.environ.get('EARTHDATA_PASSWORD')
+        EARTHDATA_EMAIL = os.environ.get('EARTHDATA_EMAIL')
```

Comment:
I'm having trouble wrapping my head around why these need to get set up as environment variables within Travis rather than just being static values we're passing into the tests/mock via `conftest.py`. The only environment value that `Earthdata.py` will attempt to look for is `EARTHDATA_PASSWORD`, so wouldn't it be better to only directly pass username and email as variables into `Earthdata()` like a user would be doing, excepting the case where they're set in the netrc file? Otherwise, I can't see where/how/why we'd expect `EARTHDATA_USERNAME` to be the same as `username` since we could easily change the former on Travis but it's not used to generate `self.uid` during the test mock setup process.

Reply:
@tsutterley Any updates on this? I've tried a few things but don't recall the intended setup well enough and fear I might be hurting rather than helping...

Reply:
Hey @JessicaS11 thanks for the reminder.  I think everything in "core" is in working condition (tested it again this morning).  However I'm having a hard time justifying the mock login tests.  They don't really test anything within the Earthdata class since they're isolated.

Reply:
Hmmm... Was the idea that we test that the correct login option is being used, which IS a valuable set of tests? Then an actual login test (independent of this PR) that pings NSIDC can verify that `Earthdata._start_session()` is working right.

Reply:
i think that's right.  i made some edits to conftest to use the same credentials in `test_behind_NSIDC_API_login.py` should we ever decide to go off the mock.  i think it should pass travis now

Reply:
Thanks @tsutterley! This looks great (and all the tests pass). Now we can move ahead with finishing up the [login docs](https://github.com/icesat2py/icepyx/tree/login_docs) page.